Documentation/Contributors/CodeReviewGuide/README.md

```diff
@@ -54,6 +54,57 @@ This guide describes best practices for code reviews.
 * Delete the branch after merging the pull request.
 * Verify that the corresponding issue (if any) was closed.
 
+## Useful Git Commit Management
+
+Sometimes it is useful to clean up the pull request. Here are some scenarios and how to tackle them.
+The tips below will use the following keywords:
+* **origin** will refer to the user fork, ie. `git@github.com/username/cesium.git`.
+* **upstream** will refer to the AGI repo, ie. `git@github.com/AnalyticalGraphicsInc/cesium.git`.
+* **mybranch** will refer to your local branch name.
+* **target** will refer to the target branch the PR is to be merged into (and also the source for `mybranch`).
+
+If you are new to git, it may be useful to create a back up of your branch in case something goes wrong.
+To do that, use:
+```
+git branch # This should show that you are currently on "mybranch", otherwise use "git checkout mybranch"
+git checkout -b mybranch-backup # This should change your current branch to "mybranch-backup", which should be identical to "mybranch"
+# if you wish to be even more conservative, you can push this to remote
+git push origin mybranch-backup
+
+# Now switch back to your working "mybranch"
+git checkout mybranch
+```
+
+### You want to squash all the commits in your pull request into a single commit
+
+```
+git reset origin/target # this will reset your branch the same as origin/target, with your changes unstaged.
```

Comment:
It's really important the you merge `origin/target` into your branch before doing the reset, otherwise the reset will undue all of the unmerged changes from `origin/target`